pkg/controller/flightctl/flightctl.go

```diff
@@ -54,45 +50,92 @@ type FlightCtlManager struct {
 	recorder     events.Recorder
 
 	flightctlClient         flightctlClient
+	flightctlServer         string
 	agentRegistrationServer string
 
 	cachedToken string
 }
 
-func (f *FlightCtlManager) ApplyResources(ctx context.Context) error {
-	var err error
+// StartReconcileFlightCtlResources starts a loop to continuously reconcile FlightCtl resources
+func (f *FlightCtlManager) StartReconcileFlightCtlResources(ctx context.Context) {
+	ticker := time.NewTicker(5 * time.Minute)
+	defer ticker.Stop()
+
+	// Helper function to apply resources and record errors
+	applyFunc := func() {
+		if err := f.ensureFlightCtlServer(); err != nil {
+			f.recorder.Event("FlightCtlServerFailed",
+				fmt.Sprintf("Failed to ensure FlightCtl server: %v", err))
+			return
+		}
 
-	// Apply kubernetes resources
-	err = f.applyKuberentesResources(ctx)
-	if err != nil {
-		return err
-	}
+		// Apply kubernetes resources
+		if err := f.applyKuberentesResources(ctx); err != nil {
+			f.recorder.Event("KubernetesResourcesFailed",
+				fmt.Sprintf("Failed to apply Kubernetes resources: %v", err))
+			return
+		}
 
-	// Create Repository resources
-	err = f.applyRepository(ctx)
-	if err != nil {
-		return err
+		// Create Repository resources
+		if err := f.applyRepository(ctx); err != nil {
+			f.recorder.Event("RepositoryFailed",
+				fmt.Sprintf("Failed to apply Repository resources: %v", err))
+			return
+		}
+
+		// Record successful sync
+		f.recorder.Event("ResourcesSynced", "Successfully synced FlightCtl resources")
 	}
 
-	return nil
+	// Run immediately for the first time
+	applyFunc()
+
+	// Then run every 5 minutes
+	for {
```

Comment:
There is wait.PollImmediateUntil so you can stop doing it after these are created.

Reply:
I did a second think, we need to keep it reconcile to keep the token fresh in flightctl Repository.

Reply:
Change to "after resources applied, keep update Repository every day"